lib/tessel/deploy.js

```diff
@@ -383,10 +397,168 @@ actions.glob = {
   }
 };
 
+actions.resolveBinaryModules = function(opts) {
+  var cwd = process.cwd();
+  var target = opts.target || cwd;
+  var relative = path.relative(cwd, target);
+  var globRoot = relative || target;
+
+  return new Promise(function(resolve, reject) {
+    // Find all modules that include a compiled binary
+    var pattern = 'node_modules/**/*.node';
+    var binaries = actions.glob.files(globRoot, [pattern]).map(function(mPath) {
+      // Garher information about each found module
+      var binName = path.basename(mPath);
+      var modulePath = binaryModulePath(path.dirname(mPath));
+      var packageJson = require(path.join(globRoot, modulePath, 'package.json'));
+      return {
+        binName: binName,
+        name: packageJson.name,
+        version: packageJson.version,
+        path: modulePath,
+        packageJson: packageJson,
+      };
+    });
+
+    // Using the discovered binaries, create a Promise to
+    // represent the request, receipt, and extraction process
+    // for each of them.
+    var requests = binaries.map(function(details) {
+      return new Promise(function(resolve, reject) {
+        var tgz = `${details.name}-${details.version}-Release.tgz`;
```

Comment:
[Template strings](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/template_strings), they are a part of ES2015 :+1: